warehouse/macaroons/services.py

```diff
@@ -135,7 +135,13 @@ def create_macaroon(self, location, user_id, description, caveats):
         """
         user = self.db.query(User).filter(User.id == user_id).one()
 
-        dm = Macaroon(user=user, description=description, caveats=caveats)
+        # NOTE: This is a bit of a hack: we keep a separate copy of the
+        # permissions caveat in the DB, so that we can display scope information
+        # in the UI.
+        permissions = next(
+            c["permissions"] for c in caveats if "permissions" in c
+        )  # pragma: no cover
```

Comment:
N.B.: I don't think there's any actual coverage missing here; I think `coverage` is getting tripped up by the `StopIteration`.